### PR TITLE
[TASK] Also sniff the code against the PSR-12 standard

### PR DESCRIPTION
**Pull request by oliverklee**
The latest version of PHP_CodeSniffer now supports PSR-12,
which supersedes PSR-2.

### CODE REVIEW

**Comment by oliverklee** [comment]
This still needs some work, it seems …

**Comment by oliverklee** [comment]
… and maybe wait for the next bug-fix release of PHP_CodeSniffer.

File: phpcs.xml.dist [hunk 1]
```diff
@@ -6,8 +6,10 @@
 
     <config name="installed_paths" value="../../slevomat/coding-standard"/>
 
-    <!--The complete PSR-2 ruleset-->
-    <rule ref="PSR2"/>
+    <!-- The complete PSR-12 ruleset (excluding a rule that requires PHP >= 7.1 -->
+    <rule ref="PSR12">
+        <exclude name="PSR12.Properties.ConstantVisibility"/>
```
Comment (JakeQZ) [inline phpcs.xml.dist]: I should have read the comment above :/.  I just wasted a few minutes trying to understand what this was about.

File: phpcs.xml.dist [hunk 2]
```diff
@@ -6,8 +6,10 @@
 
     <config name="installed_paths" value="../../slevomat/coding-standard"/>
 
-    <!--The complete PSR-2 ruleset-->
-    <rule ref="PSR2"/>
+    <!-- The complete PSR-12 ruleset (excluding a rule that requires PHP >= 7.1 -->
```
Comment (JakeQZ) [inline phpcs.xml.dist]: Missing ")" but maybe the comment for constant visibility should be separated and moved just before the relevant `<exclude>`...

File: tests/Unit/CssInlinerTest.php [hunk 3]
```diff
@@ -1606,13 +1606,18 @@ public function orderedRulesAndSurroundingCssDataProvider()
                 foreach ($possibleSurroundingCss as $descriptionAfter => $cssAfter) {
                     // every combination would be a ridiculous c.1000 datasets - choose a select few
                     // test all possible CSS before once
-                    if (($cssBetween === '' && $cssAfter === '')
+                    if (
```
Comment (JakeQZ) [inline tests/Unit/CssInlinerTest.php]: Could this be resolved by merely adding a newline after `if (`...?

File: tests/Unit/CssInlinerTest.php [hunk 4]
```diff
@@ -1606,13 +1606,18 @@ public function orderedRulesAndSurroundingCssDataProvider()
...
+                        ($cssBetween === ''
+                        && $cssAfter === '')
```
Comment (JakeQZ) [inline tests/Unit/CssInlinerTest.php]: ... I find having a newline within the bracketed `AND` expressions makes the code less readable, not more.  Is this really not allowed?

File: tests/Unit/EmogrifierTest.php [hunk 5]
```diff
@@ -1547,13 +1547,18 @@ public function orderedRulesAndSurroundingCssDataProvider()
                 foreach ($possibleSurroundingCss as $descriptionAfter => $cssAfter) {
                     // every combination would be a ridiculous c.1000 datasets - choose a select few
                     // test all possible CSS before once
-                    if (($cssBetween === '' && $cssAfter === '')
+                    if (
+                        ($cssBetween === ''
+                        && $cssAfter === '')
```
Comment (JakeQZ) [inline tests/Unit/EmogrifierTest.php]: Ditto.

File: tests/Unit/CssInlinerTest.php (hunk 3 above)
Comment (oliverklee) [reply tests/Unit/CssInlinerTest.php]: I'll test this.

This is the relevant section from the standard: https://www.php-fig.org/psr/psr-12/#51-if-elseif-else

**Comment by oliverklee** [comment]
This is the relevant section from the standard: https://www.php-fig.org/psr/psr-12/#51-if-elseif-else

File: phpcs.xml.dist (hunk 2 above)
Comment (oliverklee) [reply phpcs.xml.dist]: Done.

File: tests/Unit/CssInlinerTest.php (hunk 3 above)
Comment (oliverklee) [reply tests/Unit/CssInlinerTest.php]: Unfortunately, this did not work. :-(

Comment (oliverklee) [reply tests/Unit/CssInlinerTest.php]: We could use introduce some local variables (in a separate PR) to make the condition less enormous, though.

**Comment by oliverklee** [comment]
Updated and repushed.

File: tests/Unit/CssInlinerTest.php (hunk 3 above)
Comment (JakeQZ) [reply tests/Unit/CssInlinerTest.php]: > Unfortunately, this did not work. :-(

OK.

> We could use introduce some local variables (in a separate PR) to make the condition less enormous, though.

That would help make it more readable and perhaps also avoid the need for the comments within the `if` expression.

**Pull request merged**